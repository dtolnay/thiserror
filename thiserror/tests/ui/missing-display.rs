use thiserror::Error;

#[derive(Error, Debug)]
pub enum MyError {
    First,
    Second,
}

fn main() {}
