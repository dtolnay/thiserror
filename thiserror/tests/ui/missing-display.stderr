error[E0277]: `MyError` doesn't implement `std::fmt::Display`
 --> tests/ui/missing-display.rs:4:10
  |
3 | #[derive(Error, Debug)]
  |          ----- in this derive macro expansion
4 | pub enum MyError {
  |          ^^^^^^^ unsatisfied trait bound
  |
help: the trait `std::fmt::Display` is not implemented for `MyError`
 --> tests/ui/missing-display.rs:4:1
  |
4 | pub enum MyError {
  | ^^^^^^^^^^^^^^^^
note: required by a bound in `std::error::Error`
 --> $RUST/core/src/error.rs
  |
  | pub trait Error: Debug + Display {
  |                          ^^^^^^^ required by this bound in `Error`
  = note: this error originates in the derive macro `Error` (in Nightly builds, run with -Z macro-backtrace for more info)
